bundles/org.eclipse.cdt.lsp.clangd/src/org/eclipse/cdt/lsp/internal/clangd/ClangdFallbackManager.java

```diff
@@ -46,31 +53,40 @@ class FallbackFlags {
 	}
 
 	private static final String ISYSTEM = "-isystem"; //$NON-NLS-1$
-	private final ServiceCaller<ICBuildConfigurationManager> configManagerServiceCaller = new ServiceCaller<>(
-			getClass(), ICBuildConfigurationManager.class);
-	private final InitialFileManager initialFileManager = InitialFileManager.getInstance();
 	private final boolean isWindows = Constants.OS_WIN32.equals(Platform.getOS());
 
-	public FallbackFlags getFallbackFlagsFromInitialUri() {
+	@Reference
+	private ICBuildConfigurationManager configurations;
+	@Reference
+	private InitialUri uris;
```

Comment:
Maybe call the field only `uri` or `initialUri`, since it is only a single value